R/plotly.R

```diff
@@ -254,48 +254,64 @@ plotly_build <- function(l = last_plot()) {
   # this is ugly, but I think it is necessary, since we don't know how many 
   # traces we have until we evaluate args and call traceify() (or similar)
   x <- list()
-  for (i in seq_along(l$data)) {
-    d <- l$data[[i]]
+  x$data <- unlist(lapply(l$data, function(d) {
     if (should_eval(d)) {
       dat <- do_eval(d)
+      # put everything into a single trace
+      trace <- dat
+      trace[["indices"]] <- seq_along(trace[["x"]]) # indices of the original data elements used in the trace FIXME properly define data length
```

Comment:
@cpsievert That would make sense, but here we try to index every point in the yet-unprocessed trace before any particular splitting is conceived.